### PR TITLE
FactSet Patch - May 26

### PR DESCRIPTION
**Pull request by krithsai**
Making changes per feedback
1. Removed empty operations
2. Corrected description values
3. Removed icon and settings files

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by v-EgorKozhadei** [comment]
Hi @krithsai!

Thank you very much for submitting a new PR!

Unfortunately, you have already submitted a PR before, we need to choose which one is going to be review, this one or [the first one](https://github.com/microsoft/PowerPlatformConnectors/pull/1633), but I guess it doesn't contain the latest changes.

Please let us know which one is better for you!
You can close the PR that won't be reviewed or we can close it for you.

Please let me know if you have any questions.

**Comment by krithsai** [comment]
> Hi @krithsai!
> 
> Thank you very much for submitting a new PR!
> 
> Unfortunately, you have already submitted a PR before, we need to choose which one is going to be review, this one or [the first one](https://github.com/microsoft/PowerPlatformConnectors/pull/1633), but I guess it doesn't contain the latest changes.
> 
> Please let us know which one is better for you! You can close the PR that won't be reviewed or we can close it for you.
> 
> Please let me know if you have any questions.

@v-EgorKozhadei - Ah, my bad. We can use this PR for the purpose of review since you're right that the previous one doesn't have all the changes. Please go ahead and close the old PR.

File: certified-connectors/FactSet/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,1410 @@
...
+        "summary": "BookBuilder - Get Book List",
+        "description": "Retrieves the list of books that were previously created and are available in the client's book library",
+        "operationId": "get-book-list",
+        "parameters": []
+      }
+    },
+    "/book-builder-api/v1/template-list": {
+      "get": {
+        "responses": {
+          "200": {
```
Comment (v-EgorKozhadei) [inline certified-connectors/FactSet/apiDefinition.swagger.json:1278]: @krithsai please list all known 4xx/5xx responses for your operations if possible. You can refer to the documentation for [an example](https://docs.microsoft.com/en-us/connectors/custom-connectors/coding-standards#actions-and-triggers).

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @krithsai! Please check the comments after the review, thank you!

File: certified-connectors/FactSet/Readme.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+* `Signals - Get Headlines` : Fetch Signals event headlines based on the filtering criteria
+* `Signals - Get Details` : Fetch Signals event headlines plus all additional event details based on the filtering criteria
+* `Signals - Adaptive Card` : Fetch Microsoft's Adaptive Cards, which includes headlines and event details data plus hyperlinks to FactSet reports, based on the filtering criteria
+* `NER - Entities` : Extract named entities from document text.
+* `BookBuilder - Get Book List` : Retrieves the list of books that were previously created and are available in the client's book library
+* `BookBuilder - Get Template List` : A template is a predefined list of content to be compiled in a PDF. Templates need to defined/created in FactSet workstation.
+* `BookBuilder - Create Book from Template` : This endpoint retrieves book status, book name, and book ID for ticker requested in JSON format.
+* `BookBuilder - Get PDF` : This endpoint will return the PDF output given a book_id.
+
+### FactSet Support
```
[the remark this answers is not in the document]
Comment (krithsai) [reply certified-connectors/FactSet/Readme.md:31]: Hi @v-EgorKozhadei - The template doesn't specify that all the headers are required. Can you tell me which ones are required and missing from my submission? Besides, I checked several other certified connectors in your repo and most of them don't seem to specify every header.

File: certified-connectors/FactSet/Readme.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# FactSet Connector
```
[the remark this answers is not in the document]
Comment (krithsai) [reply certified-connectors/FactSet/Readme.md]: Made this change.

File: certified-connectors/FactSet/Readme.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# FactSet Connector
+
+Prepare for client meetings using the FactSet connector to access key company information in a custom branded, presentation ready PDF. Stay on top of relevant news and events for companies of interest and highlight important topics to drive your discussion with FactSet Signals. Alternatively, use Digital Cards to collaborate with colleagues and send summaries of actionable insights via Teams.
+
+### Publisher - FactSet Research Systems
```
[the remark this answers is not in the document]
Comment (krithsai) [reply certified-connectors/FactSet/Readme.md]: Fixed.

File: certified-connectors/FactSet/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,1410 @@
...
+                      },
+                      "summary": {
+                        "type": "string",
+                        "description": "Signal Summary"
+                      },
+                      "userRelevanceScore": {
+                        "type": "number",
+                        "format": "float",
+                        "description": "User Relevance Score",
+                        "title": ""
```
[the remark this answers is not in the document]
Comment (krithsai) [reply certified-connectors/FactSet/apiDefinition.swagger.json]: Removed property.

File: certified-connectors/FactSet/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,1410 @@
...
```
[the remark this answers is not in the document]
Comment (krithsai) [reply certified-connectors/FactSet/apiDefinition.swagger.json]: Removed property.

File: certified-connectors/FactSet/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,1410 @@
...
+                    }
+                  },
+                  "description": "errors"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Signals - Details",
+        "operationId": "get-details",
```
[the remark this answers is not in the document]
Comment (krithsai) [reply certified-connectors/FactSet/apiDefinition.swagger.json]: Fixed.

File: certified-connectors/FactSet/Readme.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (v-EgorKozhadei) [reply certified-connectors/FactSet/Readme.md:31]: Hi @krithsai! 

Not all of the headings are required.

For Verified Publishers the following headings are marked as required: 
title, publisher, prerequisites, supported operations, obtaining credentials, known issues and limitations, deployment instructions.

In [the template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md?plain=1), under each heading there's a description that says if it's necessary or not and what information it should contain.

[the remark this answers is not in the document]
Comment (krithsai) [reply certified-connectors/FactSet/Readme.md]: Made changes per feedback.

[the remark this answers is not in the document]
Comment (v-EgorKozhadei) [reply certified-connectors/FactSet/Readme.md]: Hi @krithsai, thank you for the changes! Can you please tell me what headers do you want to include and what information will be there?

**Comment by v-EgorKozhadei** [comment]
Hello @krithsai, 

I hope you are doing well. 

Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) using your latest commit id, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. 

Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.  

We expect all tests to be completed within 1-2 weeks. 

If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions. 

Please let me know if you have any questions. 

Thank you very much for working with us.

**Comment by krithsai** [comment]
Thanks for the update @v-EgorKozhadei. I will go ahead and submit to ISV Studio. You mentioned latest commit ID - is this one of these two IDs on the right side of the screenshot?

image

**Comment by v-EgorKozhadei** [comment]
@krithsai yes, the last one is about modifying the readme file.

You can also see the commit history by clicking on "Commits" tab in this PR. The one at the bottom of that page will be the latest.

Please let me know if you have any questions or need help!

**Pull request merged**